inference-engine/src/gna_plugin/gna_executable_network.hpp

```diff
@@ -67,6 +68,28 @@ class GNAExecutableNetwork : public InferenceEngine::ExecutableNetworkThreadSafe
         THROW_IE_EXCEPTION << NOT_IMPLEMENTED_str;
     }
 
+    void SetConfig(const std::map<std::string, InferenceEngine::Parameter>& config,
+                   InferenceEngine::ResponseDesc* /* resp */) override {
+        using namespace InferenceEngine::GNAConfigParams;
+        if (config.empty()) {
+            THROW_IE_EXCEPTION << "The list of configuration values is empty";
+        }
+        for (auto&& item : config) {
+            if (item.first != KEY_GNA_DEVICE_MODE) {
+                THROW_IE_EXCEPTION << "The following config value cannot be changed dynamically for ExecutableNetwork in the GNA plugin: "
+                                   << item.first << ". Only " << KEY_GNA_DEVICE_MODE << " is supported.";
+            }
+        }
+        IE_ASSERT(config.begin()->first == KEY_GNA_DEVICE_MODE);
```

Comment:
this not needed - check inside loop will handle that

Reply:
I understand this, I just wanted the express that this is expected, in case something will get broken in the code above. Let's remove if you think it's redundant.

Reply:
Fixed